actions/llm-pr-writer/action.yml

```diff
@@ -0,0 +1,146 @@
+name: llm-pr-writer
+description:
+  "This GitHub workflow harnesses the capabilities of Large Language Models
+  (LLMs) to automate the creation and enhancement of pull request (PR)
+  descriptions. Designed to streamline the development process, it significantly
+  reduces the time and effort traditionally required to draft comprehensive PR
+  descriptions, particularly for extensive changes. By meticulously analyzing
+  the modifications within a PR, the workflow generates precise, context-rich
+  summaries that facilitate improved collaboration and review efficiency."
+
+inputs:
+  # general inputs
+  checkout-repo-fetch-depth:
+    description: "number of commits to fetch"
+    required: false
+    default: "0"
+  gh-token:
+    description: "github token to make a comment with"
+    required: true
+    # will require either pull_request: write or issues: write permission
+    # https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment
+  exclude-paths:
+    description: "paths to exclude from the analysis"
+    required: false
+    default: ""
+    # Use semicolons (;) to separate multiple paths.
+    # Example: 'poetry.lock;artifacts/*'
+  openai-api-key:
+    description: "openai api key"
+    required: true
+  # grafana inputs
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+    default: llm-pr-writer
+  gc-host:
+    description: "grafana hostname"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth"
+    required: false
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
+
+    - name: Create filter string
+      id: filter
+      shell: bash
+      run: |
+        IFS=';' read -ra exclude_paths_arr <<< "${{ inputs.exclude_paths }}"
+        for i in "${!exclude_paths_arr[@]}"; do
+          exclude_paths_arr[$i]="':!${exclude_paths_arr[$i]}'"
+        done
+        echo "exclude_filter="${exclude_paths_arr[@]}"" >> "$GITHUB_OUTPUT"
+
+    - name: Genereate PR Description from Diff
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.gh-token }}
+        BRANCH: ${{ inputs.checkout-repo-fetch-depth }}
+        PR_NUMBER:
+          ${{ github.event.issue.number || github.event.pull_request.number ||
+          '' }}
+      run: |
+        # base_ref=$(gh pr view --json baseRefName | jq -r '.baseRefName')
+        # exclude_filter=$(cat $GITHUB_OUTPUT | grep exclude_filter | cut -d'=' -f2)
+        # diff_command="git diff origin/$base_ref...HEAD -- $exclude_filter"
+        # echo "$diff_command"
+        # diff_output=$(eval $diff_command)
```

Comment:
I was using `gh pr diff` before. However they currently don't support file exclusion filters and this sometimes resulted in build artifacts being sent as a part of the prompt, causing it to cross the token limit threshold.
We had to go with the elaborate ways of computing the branch split to head and resort to `git diff` to get this functionality.

Reply:
ahh the reason I commented it out because of this error here:
https://github.com/smartcontractkit/falcon_llm/actions/runs/8180659211/job/22369114574?pr=6#step:2:168

it's about finding the ref for some reason yet it works with the `gh` way.

Reply:
so it seems like why that error appeared is due to this line being removed:
```
gh pr checkout ${{ github.event.issue.number || github.event.pull_request.number || '' }}
```
As some how gh CLI needs to check out the PR again (even after running action/checkout) to have other metadata generated for it.